synapse/handlers/room.py

```diff
@@ -732,6 +735,14 @@ async def create_room(
         invite_3pid_list = config.get("invite_3pid", [])
         invite_list = config.get("invite", [])
 
+        # validate each entry
+        for invite_3pid in invite_3pid_list:
+            try:
+                assert has_3pid_invite_keys(invite_3pid)
```

Comment:
`assert` is unsuitable for tests like this. It's a debugging aid which is optimised out when python is run with `-O`. (Also, it raises an AssertionError which will be translated into an HTTP 500 error, which is inappropriate for an invalid request).